flink-runtime/src/main/java/org/apache/flink/runtime/rpc/resourcemanager/ResourceManager.java

```diff
@@ -52,18 +61,45 @@
 public class ResourceManager extends RpcEndpoint<ResourceManagerGateway> {
 	private final ExecutionContext executionContext;
 	private final Map<JobMasterGateway, InstanceID> jobMasterGateways;
+	private final Map<ResourceID, TaskExecutorGateway> taskExecutorGateways;
+	private final Map<ResourceID, ResourceManagerToTaskExecutorHeartbeatScheduler> heartbeatSchedulers;
+	private final LeaderElectionService leaderElectionService;
+	private UUID leaderSessionID;
+	// TODO private final SlotManager slotManager;
```

Comment:
SlotManager is doing by kurt. So I marked some TODO here waiting for his PR. It would be removed after SlotManager code committed